emu_mps/mps_backend_impl.py

```diff
@@ -510,6 +540,45 @@ def fill_results(self) -> None:
 
                 callback(self.config, fractional_time, full_state, full_mpo, self.results)
 
+    def permute_results(self, results: Results) -> Results:
+        if not torch.equal(
+            self.qubit_permutation, optimat.eye_permutation(self.qubit_count)
+        ):
+            inv_perm = optimat.inv_permutation(self.qubit_permutation)
+            permute_bitstrings(results, inv_perm)
+            permute_occupations_and_correlations(results, inv_perm)
+            permute_atom_order(results, inv_perm)
+        return results
+
+
+def permute_bitstrings(results: Results, perm: torch.Tensor) -> None:
```

Comment:
those need to be unit-tested

Reply:
the more small, short, good unit tests you have, the less messy, complicated integration tests you have to write

Reply:
done